### DISCUSSION TITLE
code-server 3.12.0 coming soon!

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#4178] [Announcements]
We published v3.12.0 earlier today and had a few hiccups and are in the process of manually releasing on all platforms:

- ✅ [npm](https://www.npmjs.com/package/code-server)
- ✅ [aur](https://aur.archlinux.org/packages/code-server/)
- ✅ [Homebrew](https://formulae.brew.sh/formula/code-server#default) 
- ✅ [Docker](https://registry.hub.docker.com/r/codercom/code-server/tags?page=1&ordering=last_updated)   

We realized there was an issue with the cross-compilation step. We believe we fixed it [here](https://github.com/cdr/code-server/pull/4187) and re-uploaded the `.deb` and the `.rpm` assets to the `3.12.0` release. Please download from there.

**What are you doing to prevent future hiccups?**

[3.12.1](https://github.com/cdr/code-server/milestone/30) will only focus on improving the release process so that future releases go more smoothly.

**Comment by nooobcoder** [comment]
@jsjoeio out of the release changes, i could not understand this
 - Migrated to vscode fork

What does this mean? I knew `code-server` is a fork of VS Code from Microsoft, then what this forking means now.
Please explain this to me like I am 8.

Thank you,
Hope you are having a great day.

**Reply by lg-nineoneninethreenine** [reply]
Ah, I understood! Somebody from the @cdr organisation _forked_ microsoft/vscode, so they migrated to _their_ fork from an older version. Am I correct @jsjoeio?

**Reply by jsjoeio** [reply]
You both are super close! Basically, we did have a fork of VS Code from Microsoft that lived under `lib/vscode` as a git subtree (used to be a git submodule back before my time).

We moved it outside of the code-server codebase. Now, we technically have two repos:
- the fork of microsoft/vscode as @im-coder-lg pointed out
- code-server (this repo)

We have a module inside code-server that connects the two using `yarn`. @TeffenEllis will eventually put out documentation to better explain how it fits together but that's the gist :)
